prime-router/src/main/kotlin/history/SubmissionHistory.kt

```diff
@@ -512,15 +512,13 @@ class DetailedSubmissionHistory(
         }
 
         if (destinations.size == 0) {
-            /**
-             * Cases where this may hit:
-             *     1) Data hasn't been processed yet (common in async submissions)
-             *     2) Very rare: No data matches any geographical location.
-             *         e.g. If both the testing tab and patient data were foreign addresses.
-             * At the moment we have NO easy way to distinguish the latter rare case,
-             * so it will be treated as status RECEIVED as well.
-             */
-            return Status.RECEIVED
+            return if (reportItemCount != 0) {
+                // Data hasn't been processed yet (common in async submissions)
+                Status.RECEIVED
+            } else {
```

Comment:
In testing, I still can't hit this else case to hit this new `NOT_DELIVERED`, seems like the `reportItemCount` is a count of how many items are in the report, not of how many receivers get the report, so don't think this works quite right.

Reply:
I'm going to circle back with Luis to see what he was using to test this when he wrote the original comment as I'm having a hard time hitting this condition as well outside of tests.

Reply:
I think the logic behind testing `reportItemCount` is the change made for #9224 in that the item count is now explicitly set to zero when writing the "none" report data during routing. I'm not 100% sure where the item count being checked here is sourced from though.

I was also having difficulty reproducing the original error. Another change made in the time since the bug was written might have changed the originally observed behavior.